SI-6799 Adds option to trace when a symbol from a list of ids is created...

JamesIry opened the pull request:
....

Create -Ytraceids-debug, a debugging aid to find out where
specified symbols are created by dumping  a stack trace.
This change also meant adding a settings type for multiple ints.

Commit:
SI-6799 Adds option to trace when a symbol from a list of ids is created.

Create -Ytraceids-debug, a debugging aid to find out where
specified symbols are created by dumping  a stack trace.
This change also meant adding a settings type for multiple ints.

JamesIry commented:
review @paulp

xeno-by commented on src/reflect/scala/reflect/internal/Symbols.scala:
```diff
@@ -139,7 +139,11 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
     rawatt = initPos
 
     val id = nextId() // identity displayed when -uniqid
-    //assert(id != 3390, initName)
```
Can't believe this marvel is about to be deleted.

JamesIry replied:
Yeah, whoever was clearly doing what I'm doing with this patch except...painfully.

paulp replied:
Agreed, this demands further elaboration. And where is the test case to prevent regression on symbols with id #3390? I'm a little disappointed in this pull request.

JamesIry replied:
I'm abashed at my own behavior here. I shall never delete another commented assertion again without a SIP.

VladUreche replied:
Pease leave a comment there for people who will still be inclined to use the assert: "don't make it more painful than it needs to be: use -Ytraceids-debug".

xeno-by commented on src/compiler/scala/tools/nsc/settings/MutableSettings.scala:
```diff
@@ -526,7 +527,32 @@ class MutableSettings(val errorFn: String => Unit)
+  /** A setting that accumulates all ints supplied to it,
+   *  until it encounters one starting with a '-'. Because '-' is special
+   *  negative ints must be represented using '~', e.g. ~12 is -12*/
+  class MultiIntSetting private[nsc](
```
Would it be possible to reduce duplication w.r.t `MultiStringSetting`?

JamesIry replied:
https://github.com/JamesIry/scala/commit/7e672c3785a16764a1e897739517a0909a600cde

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1857/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1857/

scala-jenkins commented:
Started jenkins job pr-rangepos at https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/1143/

scala-jenkins commented:
jenkins job pr-rangepos: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/1143/

Commit:
SI-6799 Remove duplication between MultiIntSetting and MultiStringSetting

MultiIntSetting and MultiStringSetting are basically the same thing
with a slight difference in how they parse their arguments.
This commit reflects that similarity.

JamesIry commented:
Won't fly past the peanut gallery.

VladUreche commented:
James, in case you decide to resurrect this PR (which seems like a great idea to me) please also add tracing to the info setter -- I often find myself asking "who set the wrong type for this symbol" -- and I need to hijack the setter to find out.
